src/app/EventManagement.cpp

```diff
@@ -460,8 +460,8 @@ CHIP_ERROR EventManagement::CopyAndAdjustDeltaTime(const TLVReader & aReader, si
 
     if (aReader.GetTag() == TLV::ContextTag(to_underlying(EventDataIB::Tag::kPath)))
     {
-        err =
-            ctx->mpWriter->Put(TLV::ContextTag(to_underlying(EventDataIB::Tag::kEventNumber)), ctx->mpContext->mCurrentEventNumber);
+        err = ctx->mpWriter->Put(TLV::ContextTag(to_underlying(EventDataIB::Tag::kEventNumber)),
+                                 ctx->mpContext->mCurrentEventNumber - 1);
```

Comment:
Please document why this random `- 1` is here.  Why is `ctx->mpContext->mCurrentEventNumber` not the right number?

And separately, _can_ it be made the right number?

Reply:
yes, mCurrentEventNumber is increased to next Event Number when it found it is same priority with the one set by FetchEventSince.  After that, we start to encoding, where we need -1,   later mCurrentEventNumber is returned to report engine which is used as next starting event number.

Reply:
Just introduce bool variable, then after finish encoding, we check if this boolean is true, then increase the event number for next use.

Reply:
Done, thanks